### PR TITLE
Editorial: add [[AgentSignifier]] to Realm Record

### LINKED ISSUE #1357: Define Agent (and Agent Cluster) more concretely
As @allenwb wrote in https://github.com/tc39/ecma262/issues/882#issuecomment-293334655 it'd be good to formally define agent as holding:

> * A set of Realms.  The realms in the set are private to the Agent and may not be accessible to any other agent.
> * A heap of SharedArrayBuffer objects that whose access may be shared with other Agents
> * A set of [well-known symbol values](https://tc39.github.io/ecma262/#sec-well-known-symbols)
> * A [GlobalSymbolRegistry](https://tc39.github.io/ecma262/#sec-symbol.for)

It would also be good to formalize how to get from a Realm to its corresponding Agent (e.g., a language construct a host could reuse).

It would also be good to have the same for agent clusters. I.e., how to get to the cluster from an object (possibly via the realm or agent), and what the cluster consists of (a set of agents).

This is especially useful for hosts that enable communication between agent clusters (i.e., HTML).

(This is effectively what #882 morphed into, but restated for clarity and brevity.)

### PR DESCRIPTION
**Pull request by syg**

### CODE REVIEW

**Comment by syg** [comment]
I contend this is editorial because a lot of things wouldn't make sense if realms don't belong to exactly one agent: things like [[KeptAlive]] being able to change under the hood would be bad.

File: spec.html [hunk 1]
```diff
@@ -12003,11 +12015,11 @@ <h1>InitializeHostDefinedRealm ( ): either a normal completion containing ~unuse
...
     <emu-note>
       <p>Some web browsers share a single executing thread across multiple unrelated tabs of a browser window, for example.</p>
     </emu-note>
-    <p>While an agent's executing thread executes jobs, the agent is the <dfn id="surrounding-agent" variants="surrounding agents">surrounding agent</dfn> for the code in those jobs. The code uses the surrounding agent to access the specification-level execution objects held within the agent: the running execution context, the execution context stack, and the Agent Record's fields.</p>
+    <p>While an agent's executing thread is executing algorithmic steps, the agent is the <dfn id="surrounding-agent" variants="surrounding agents">surrounding agent</dfn> for those steps. The code uses the surrounding agent to access the specification-level execution objects held within the agent: the running execution context, the execution context stack, and the Agent Record's fields.</p>
```
Comment (jmdyck) [inline spec.html:12022]: You might want to eliminate the other use of "code" in this para.

**Comment by jmdyck** [comment]
> Fixes #1357

The first post of that issue is asking for more than what this PR provides.

**Comment by syg** [comment]
> Fixes https://github.com/tc39/ecma262/issues/1357

I think the realm -> agent pointer is the most glaring thing that ought to be fixed, I'd rather think through the other parts like the symbol registry afterwards. I can remove the "Fixed #1357" bit from the commit message.

**Review by michaelficarra — APPROVED** [review]

File: spec.html (hunk 1 above)
Comment (syg) [reply spec.html:12022]: Done.

**Comment by syg** [comment]
@bakkot PTAL, small pre-req PR for `Atomics.waitAsync`.

**Review by bakkot — APPROVED** [review]

**Pull request merged**